test/collect-reward-test.js

```diff
@@ -0,0 +1,81 @@
+process.env.NODE_ENV = 'test';
+const request = require('supertest');
+const { expect } = require('chai');
+const app = require('../src/index');
+const jwt = require('jsonwebtoken');
+
+let newAccTokenHeader = '';
+
+describe('POST /login', () => {
+  it('should get back the userId from the accessToken and the header for POST /hero/use tests', (done) => {
+    request(app)
+      .post('/login')
+      .send({
+        username: 'Daniel',
+        password: 'xxxx5555',
+      })
+      .end((err, data) => {
+        if (err) return done(err);
+        newAccTokenHeader = `Bearer ${data.body.accessToken}`;
+        expect(JSON.stringify(jwt.decode(data.body.accessToken).userId)).to.equal(JSON.stringify(1));
+        return done();
+      });
+  });
+});
+
+describe('PUT /collect', () => {
+  it('should respond with added equipment', (done) => {
+    request(app)
+      .put('/collect')
+      .set('Authorization', newAccTokenHeader)
+      .set('heroid', '1')
+      .end((err, data) => {
+        if (err) return done();
+        expect(JSON.stringify(data.body)).to.equal(JSON.stringify({
+          changed: [],
+          added: [{
+            id: 394,
+            name: 'Sword of minor bullshit',
+            type: 'LeftHand',
+            active: 0,
+          }],
+          removed: [],
+        }));
+        return done();
+      });
+  });
+  it('should respond with error message if dungeon not yet finnised', (done) => {
+    request(app)
+      .put('/collect')
+      .set('Authorization', newAccTokenHeader)
+      .set('heroId', '4')
+      .set('Accept', 'application/json')
+      .end((err, data) => {
+        if (err) return done(err);
+        expect(JSON.stringify(data.body)).to.equal(JSON.stringify('Dungeon not finnished yet'));
+        return done();
+      });
+  });
+  it('should respond with error message if heroId missing', (done) => {
+    request(app)
+      .put('/collect')
+      .set('Authorization', newAccTokenHeader)
+      .set('Accept', 'application/json')
+      .end((err, data) => {
+        if (err) return done();
+        expect(JSON.stringify(data.body)).to.equal(JSON.stringify('Please enter a valid heroId'));
+        return done();
+      });
+  });
+  it('should respond with error message if heroId not num', (done) => {
+    request(app)
+      .put('/collect')
+      .set('Authorization', newAccTokenHeader)
+      .set('heroid', 'asd')
+      .end((err, data) => {
+        if (err) return done();
+        expect(JSON.stringify(data.body)).to.equal(JSON.stringify('Please enter a valid heroId'));
+        return done();
+      });
+  });
+});

```

Comment:
Same problem i have in my earlier backend task that I still need to fix. You don't need a different describe test to login. You can do it in the exact describe test as a before() function

Reply:
corrected, BUT - I think if there is a task for it, then it should have been okay to leave it as it was